tsdb/store.go

```diff
@@ -1709,67 +1709,87 @@ func (s *Store) TagValues(ctx context.Context, auth query.Authorizer, shardIDs [
 			switch e.Op {
 			case influxql.EQ, influxql.NEQ, influxql.EQREGEX, influxql.NEQREGEX:
 				tag, ok := e.LHS.(*influxql.VarRef)
-				if !ok || tag.Val != "_name" {
+				if !ok || tag.Val != measurementKey {
 					return nil
 				}
 			}
 		}
 		return e
 	}), nil)
 
-	filterExpr := influxql.CloneExpr(cond)
-	filterExpr = influxql.Reduce(influxql.RewriteExpr(filterExpr, func(e influxql.Expr) influxql.Expr {
+	is, err := s.indexSetFromShards(ctx, shardIDs)
```

Comment:
Getting the index set was refactored into a separate method so that both the Flux and InfluxQL path can use it, but its behavior is unchanged.